fem/tmop_tools.cpp

```diff
@@ -523,8 +526,12 @@ real_t TMOPNewtonSolver::ComputeScalingFactor(const Vector &x,
       avg_fit_err = 0.0;
       max_fit_err = 0.0;
 
+      //
       // Update the mesh and get the L-vector in x_out_loc.
-      add(x, -scale, c, x_out);
+      //
+      // Form x_out = x_0 + (d_in - scale * c).
+      add(d_in, -scale, c, d_out);
+      add(x_0, d_out, x_out);
```

Comment:
But we have `x_in = x_0+d_in` already,

Reply:
Yes it can be `add(x_in, -scale, c, x_out)`, but both `d_out` and `x_out` are needed below. I improved the comment.